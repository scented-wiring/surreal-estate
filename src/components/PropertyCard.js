import React from "react";
import "../styles/PropertyCard.css";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBath,
  faBed,
  faPoundSign,
  faEnvelope,
  faStar,
} from "@fortawesome/free-solid-svg-icons";

const PropertyCard = ({
  _id,
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
  userID,
  onSaveProperty,
}) => {
  return (
    <div className="property-card-display">
      <div className="property-title">{title}</div>
      <div className="property-type-city">
        {type}, {city}
      </div>
      <div className="property-bathrooms">
        <FontAwesomeIcon icon={faBath} /> {bathrooms}
      </div>
      <div className="property-bedrooms">
        <FontAwesomeIcon icon={faBed} /> {bedrooms}
      </div>
      <div className="property-price">
        <FontAwesomeIcon icon={faPoundSign} />
        {price}
      </div>
      <button className="button">
        <a className="button-text" href={"mailto:" + email}>
          <FontAwesomeIcon icon={faEnvelope} /> Email
        </a>
      </button>
      {userID && (
        <button className="button" onClick={() => onSaveProperty(_id)}>
          <FontAwesomeIcon icon={faStar} /> Save
        </button>
      )}
    </div>
  );
};

PropertyCard.propTypes = {
  _id: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  bathrooms: PropTypes.string,
  bedrooms: PropTypes.string,
  price: PropTypes.string,
  city: PropTypes.string,
  email: PropTypes.string,
  userID: PropTypes.string,
  onSaveProperty: PropTypes.func.isRequired,
};

export default PropertyCard;
