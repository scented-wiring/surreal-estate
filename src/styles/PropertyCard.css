.property-card-display > div {
  margin: 0 0 10px 0;
}

.property-title {
  font-size: 18px;
  font-weight: bold;
}

.property-type-city {
  font-style: italic;
}

.button {
  background-color: purple;
  color: white;
  border: none;
  width: 100%;
  margin-bottom: 10px;
  margin-left: auto;
  margin-right: auto;
  padding: 5px 0 5px 0;
}

.button:hover {
  background-color: #363636;
  transition: all 0.4s ease 0s;
}

button > a {
  color: white;
}

button > a:hover {
  color: white;
  font-weight: normal;
}
