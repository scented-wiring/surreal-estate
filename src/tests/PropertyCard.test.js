import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const mockData = {
    _id: "000000",
    title: "2 Bedroom Flat",
    type: "Flat",
    bathrooms: "1",
    bedrooms: "2",
    price: "1000",
    city: "Manchester",
    email: "example@test.com",
    userID: "11111",
    onSaveProperty: jest.fn(),
  };

  it("renders a component", () => {
    const { asFragment } = render(<PropertyCard />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a title", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("2 Bedroom Flat")).toHaveClass("property-title");
  });

  it("renders a type and city", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("Flat, Manchester")).toHaveClass("property-type-city");
  });

  it("renders a bathroom number", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("1")).toHaveClass("property-bathrooms");
  });

  it("renders a bedroom number", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("2")).toHaveClass("property-bedrooms");
  });

  it("renders a price", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("1000")).toHaveClass("property-price");
  });

  it("renders an email button", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("Email")).toHaveClass("button-text");
  });

  it("renders a save button if user is logged in", () => {
    const { getByText } = render(<PropertyCard {...mockData} />);

    expect(getByText("Save")).toHaveClass("button");
  });

  it("doesn't render a save button if user is logged out", () => {
    const { queryByText } = render(<PropertyCard userID="" />);

    expect(queryByText("Save")).not.toBeInTheDocument();
  });
});
