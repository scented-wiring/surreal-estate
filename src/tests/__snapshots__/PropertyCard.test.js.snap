// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders a component 1`] = `
<DocumentFragment>
  <div
    class="property-card-display"
  >
    <div
      class="property-title"
    />
    <div
      class="property-type-city"
    >
      , 
    </div>
    <div
      class="property-bathrooms"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bath fa-w-16 "
        data-icon="bath"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          fill="currentColor"
        />
      </svg>
       
    </div>
    <div
      class="property-bedrooms"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed fa-w-20 "
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          fill="currentColor"
        />
      </svg>
       
    </div>
    <div
      class="property-price"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pound-sign fa-w-10 "
        data-icon="pound-sign"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
          fill="currentColor"
        />
      </svg>
    </div>
    <button
      class="button"
    >
      <a
        class="button-text"
        href="mailto:undefined"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
         Email
      </a>
    </button>
  </div>
</DocumentFragment>
`;
